src/Data/Lens/Prism.purs

```diff
@@ -18,23 +57,78 @@ import Data.Profunctor (dimap, rmap)
 import Data.Profunctor.Choice (right)
 import Data.Newtype (under)
 
--- | Create a `Prism` from a constructor/pattern pair.
+-- | Create a `Prism` from a constructor and a "focus" function that
+-- | produces an `Either`:
+-- | 
+-- | ```purescript
+-- | solidFocus :: Prism' Fill Color
+-- | solidFocus = prism Solid case _ of
+-- |   Solid color -> Right color
+-- |   anotherCase -> Left anotherCase
+-- | ```
 prism :: forall s t a b. (b -> t) -> (s -> Either t a) -> Prism s t a b
 prism to fro pab = dimap fro (either id id) (right (rmap to pab))
 
+-- | Create a `Prism` from a constructor and a "focus" function that
```

Comment:
What do you think about calling this a matching function? I compare it to pattern matching on the constructor.

Reply:
I'll do that.